docs/metrics.md

```diff
@@ -0,0 +1,440 @@
+
+-   The setup flow, so we can know at what points (if any) people quit the flow before finishing it
+-   Top-level interactions centered around use of the Lockbox toolbar icon. This includes interactions within the initial pop-over that is displayed when the user clicks the icon.
+-   Interactions with the main CRUD editor view which lists the user's Lockbox items (credentials)
+-   Interactions with the add / modify dialogs used to enter / edit item information
+-   Changes to the datastore that actually contains the user's items, in addition to user actions that lead to those changes
+-   When the user submits feedback about lockbox
+
+Each of these are described below within their own Events subsection.
+
+
+## Non-Event Metrics
+
+These are the metrics we plan to collect regarding the state of user datastores. Note that we won't be able to record these directly for alpha. We will have to infer them from the event data.
+
+-   `n_items` The number of credentials that exist in the user's datastore. Integer
+
+-   `n_notes` The number of items for which the user has manually entered custom notes for. Integer
+-   `timestamp_last` The timestamp of the last edit the user made to the datastore. Does not necessarily correspond to the last time they opened the CRUD editor.
+
+## Events
+
+### Setup Events
+
+From a metrics point of view, the FxA sign-in / account creation process is a black box to us. Thus the best we can do at this point is track when a user interacts with the Lockbox-specific views that are included in the first run flow. We have two **categories** of events here: rendering events and interaction events:  `setup.render` and `setup.interaction`. For rendering events, the **object** field is populated by the name of the view being rendered; for interaction events the **object** is the name of the button. Below is a brief description of the view names and their corresponding buttons.
+
+-   `lockbox_toolbar_firstrun`
+    -   Pop-over shown when the user first clicks on the Lockbox toolbar icon after install. Contains the `signin_button`. Note that the click on the toolbar icon itself is logged in a separate event (see "Toolbar Button Interaction Event" section below).
+-   `welcome`
+    -   Shown after sign-in button is clicked. Contains the `fxa_login_button`
+-   `confirm_master_password`
+    -   Shown after FxA process is completed. Prompts user to re-use their FxA pw as their lockbox master pw. Contains the `confirm_master_pw_button`
+-   `done`
+    -   Final view in setup flow. Contains `open_button` that opens the item editor.
+
+
+
+The **methods** for these events are `render` and `click` for rendering and interaction events, respectively.
+
+
+The **extra** field contains the user's FxA anonymous/random user id. For `lockbox_toolbar_firstrun` and `welcome` views this will be `null`.
+
+
+To register the rendering events:
+```javascript
+Services.telemetry.registerEvents("setup.render", {
```

Comment:
Do we want the event category to have "lockbox" in it somewhere? That would probably avoid confusion.